src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetProjectManagerService.cs

```diff
@@ -142,6 +183,36 @@ public async ValueTask<IProjectContextInfo> GetProjectAsync(string projectId, Ca
             return results.ToArray();
         }
 
+        // Get target framework information for BuildIntegratedNuGetProjects
+        public async ValueTask<IReadOnlyCollection<TargetFrameworkInformation>> GetTargetFrameworksAsync(
+            string projectId,
+            CancellationToken cancellationToken)
+        {
+            Assumes.NotNullOrEmpty(projectId);
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            NuGetProject? project = await SolutionUtility.GetNuGetProjectAsync(
+                _sharedState.SolutionManager,
+                projectId,
+                cancellationToken);
+
+            Assumes.NotNull(project);
+
+            if (project is BuildIntegratedNuGetProject buildIntegratedProject)
+            {
+                var dgcContext = new DependencyGraphCacheContext();
+                var packageSpecs = await buildIntegratedProject.GetPackageSpecsAsync(dgcContext);
```

Comment:
For LegacyPackageReferenceProject, this will cause a UI thread switch. Given that for a LegacyPackageReferenceProject the framework is known at creation time, I'd recommend you add an abstraction you can call so that we don't end up going to the UI thread unnecessarily. 

I think it's a combination of your old implementation + new one. 

@zivkan thoughts?

Reply:
Bringing this back up to the top :)

Reply:
Changed this to save targetframework when LegacyPackageReferenceProject is created.